core/dbt/events/base_types.py

```diff
@@ -88,6 +102,18 @@ def get_pid(self) -> int:
             self.pid = os.getpid()
         return self.pid
 
+    def get_node_info(self):
+        return None
+
+
+class NodeInfo(Event, metaclass=ABCMeta):
```

Comment:
it also looks like you're using `@dataclass` notation, so you should add the decorator. You'll need `# type: ignore` because of a mypy bug on using that decorator on ABCs.